### PR TITLE
Add Testing for the Add and Remove tasks

### PR DESCRIPTION
**Pull request by Danieldotcomcoder**
Added pure functions for Add and Remove tasks
Added tests to check the functionality of Add/Remove functions.
Added Mock HTML
Added Mock LocalStorage

To check The Tests type : npm start test in terminal .

### CODE REVIEW

File: src/__tests__/function.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { addToDo, removeToDo } from '../purefunctions.js';
+
+test('Test Check', () => {
+  expect(true).toBe(true);
+});
+
+describe('Testing Adding and Removing tasks', () => {
```
Comment (rloterh) [inline src/__tests__/function.test.js:7]: - [ ] Other _pure functions_ in your code are **not tested** aside _add item_ and _delete item_. Kindly ensure to test **all** pure functions such as the _edit_, _complete: true/false_, _clear_ all completed_ and other created pure functions.

File: src/__tests__/function.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { addToDo, removeToDo } from '../purefunctions.js';
+
```
Comment (rloterh) [inline src/__tests__/function.test.js:2]: - [ ] There are some test cases that **failing**. Kindly ensure that all test cases are passing.

**Failing jest test:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi  Danny & Munsa, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Test cases are passings
- Test covers **add item** and **delete item** functions
- Tests are grouped using the `description()` method


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/function.test.js (hunk 1 above)
Comment (Danieldotcomcoder) [reply src/__tests__/function.test.js:7]: Thank you , but in the requirments for this project we should test the add and delete item  only

File: src/__tests__/function.test.js (hunk 2 above)
Comment (Danieldotcomcoder) [reply src/__tests__/function.test.js:2]: Thank you , the tests are working on my side 
image

Comment (Munsa1) [reply src/__tests__/function.test.js:2]: Get Started - TO-DO-LIST - Visual Studio Code 11_4_2021 2_31_24 PM


It's working for me too.

File: src/__tests__/function.test.js (hunk 1 above)
Comment (henatan99) [reply src/__tests__/function.test.js:7]: - [ ] May not be exactly as the previous reviewer suggests, but you need to add tests for all functions related to adding and removing items. I can see you have used pure functions as mock functions for the unit tests. But the requirement says you should use mock functions for localStorage and DOM. I suggest you import the original 'createNewTodo` and `deleteTodo` functions from `managelists.js` module file into the test file and use localStorage and DOM mock functions. For your reference I put a section of the requirements below:

```
Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:
- Mock a storage object to "imitate" localStorage operations.
- Mock HTML to test if add/delete functions add or remove exactly one <li> element to/from the list in the DOM.
```

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Danieldotcomcoder , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by henatan99 — APPROVED** [review]
Hi @Danieldotcomcoder , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**